pkg/collector/check/logreceiver.go

```diff
@@ -0,0 +1,64 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package check
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/comp/logs/integrations/def"
+)
+
+// checkContext holds a list of reference to different components used by Go and Python checks.
+//
+// This is a temporary solution until checks are components themselves and can request dependencies through FX.
+//
+// This also allows Go function exported to CPython to recover there reference to different components when coming out
+// of C to Go. This way python checks can submit metadata to inventorychecks through the 'SetCheckMetadata' python
+// method.
+type logReceiver struct {
+	lr integrations.Component
+}
+
+var logRcv logReceiver
+var logReceiverMutex = sync.Mutex{}
+
+// GetLogsReceiver returns a reference to the logs_receiver component for Python and Go checks to use.
+func GetLogsReceiver() (integrations.Component, error) {
+	logReceiverMutex.Lock()
+	defer logReceiverMutex.Unlock()
+
+	if logRcv.lr == nil {
+		return nil, errors.New("logsReciever context was not set")
+	}
+
+	return logRcv.lr, nil
+}
+
+func LogsReceiverSendLog(log, logID string) {
```

Comment:
Do you need both `LogsReceiverSendLog` and `GetLogsReceiver`? I expect you can just use `GetLogsReceiver` whenever you need to send a log. 

(If you plan to keep this function - you need to `defer` the `logReceiverMutex.Unlock()` call)

Reply:
It's not strictly necessary but I think from the context in which it would be called, it is a little more clear than assigning a variable to the result from `GetLogsReceiver` and then calling `sendLog` on that variable. I'm open to removing it however if it's unnecessary.

Reply:
I guess the question then becomes how to deal with error handling. Because right now `LogsReceiverSendLog` doesn't check if `lr` is `nil`. If the expected behavior is to silently drop logs when `lr` doesn't exist then this is fine (and you can probably remove the error handling elsewhere). 

But we probably want to generate an error log in this case. So you may want to add handling here as well. 

Either way - I assume you only need one of the two functions so feel free to remove whichever one is unused.